sycl/source/context.cpp

```diff
@@ -37,13 +37,26 @@ context::context(const vector_class<device> &DeviceList,
   if (DeviceList.empty()) {
     throw invalid_parameter_error("DeviceList is empty.");
   }
-  if (DeviceList[0].is_host())
-    impl = std::make_shared<detail::context_impl>(DeviceList[0], AsyncHandler);
-  else
-    // TODO also check that devices belongs to the same platform
-    impl = std::make_shared<detail::context_impl>(DeviceList, AsyncHandler);
+  auto number = std::find_if_not(
+      DeviceList.begin(), DeviceList.end(),
+      [&](const device &current_device) { return current_device.is_host(); });
+  if ( number == DeviceList.end() && DeviceList.back().is_host())
+    impl =
+        std::make_shared<detail::context_impl>(DeviceList[0], AsyncHandler);
+  else {
+    const device reference_device = *number;
+    const auto reference_platform = reference_device.get_platform().get();
+    if (std::any_of(DeviceList.begin(), DeviceList.end(),
+                    [&](const device &current_device) {
+                      return current_device.get_platform().get() !=
```

Comment:
Have you tested the case when user passes both actual devices and host device? From what I see, this list still can contain host device and this code might cause exception/crashes, because `.get()` doesn't work for host devices